### PR TITLE
darling: init at Dec 8, 2015

### PR DESCRIPTION
**Pull request by fxfactorial**

### CODE REVIEW

File: pkgs/tools/filesystems/darling-dmg/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{ stdenv, fetchgit, lib, cmake, fuse, zlib, bzip2, openssl, libxml2, icu } :
+
+stdenv.mkDerivation rec {
+  name = "darling-dmg";
+
+  src = fetchgit {
```
Comment (copumpkin) [inline pkgs/tools/filesystems/darling-dmg/default.nix]: `fetchFromGitHub` is far more efficient than `fetchgit`

File: pkgs/tools/filesystems/darling-dmg/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    url = git://github.com/darlinghq/darling-dmg.git;
+    rev = "9522e3907b82f6cde141b3e0e0063f09c8710cbf";
+    sha256 = "5d968b0609f9bfbecc26753d21b5182e15183b672967f7dec6038404cd6a6d7f";
+  };
+  buildInputs = [ cmake fuse openssl zlib bzip2 libxml2 icu ];
+  cmakeConfigureFlagFlags = ["-DCMAKE_BUILD_TYPE=RELEASE"];
+  meta = {
+    homepage = http://www.darlinghq.org/;
+    description = lib.strings.concatStringsSep " "
```
Comment (copumpkin) [inline pkgs/tools/filesystems/darling-dmg/default.nix]: I think the guidelines try to keep it to a single short sentence, if you can.

File: pkgs/tools/filesystems/darling-dmg/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{ stdenv, fetchgit, lib, cmake, fuse, zlib, bzip2, openssl, libxml2, icu } :
+
+stdenv.mkDerivation rec {
+  name = "darling-dmg";
+
+  src = fetchgit {
+    url = git://github.com/darlinghq/darling-dmg.git;
+    rev = "9522e3907b82f6cde141b3e0e0063f09c8710cbf";
```
Comment (copumpkin) [inline pkgs/tools/filesystems/darling-dmg/default.nix]: Is the rev attached to a tag? Could that tag be included in the `name` above?

**Comment by copumpkin** [comment]
Thanks! Our commit message guidelines would prefer the subject of the commit to be something like "darling: init at <version>"

File: pkgs/tools/filesystems/darling-dmg/default.nix (hunk 2 above)
Comment (fxfactorial) [inline pkgs/tools/filesystems/darling-dmg/default.nix]: Its not my description, straight from their website. Didn't know how much liberty I had with it.

File: pkgs/tools/filesystems/darling-dmg/default.nix (hunk 3 above)
Comment (fxfactorial) [inline pkgs/tools/filesystems/darling-dmg/default.nix]: No tag.

File: pkgs/tools/filesystems/darling-dmg/default.nix (hunk 1 above)
Comment (fxfactorial) [inline pkgs/tools/filesystems/darling-dmg/default.nix]: Okay, will have to look at it. (My 3rd real day with nix heh)

Comment (copumpkin) [inline pkgs/tools/filesystems/darling-dmg/default.nix]: Thanks, it generally looks pretty food!

**Comment by fxfactorial** [comment]
@copumpkin Sorry for delay, can you take a gander at this again, thanks.

File: pkgs/tools/filesystems/darling-dmg/default.nix [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    repo = "darling-dmg";
+    rev = "9522e3907b82f6cde141b3e0e0063f09c8710cbf";
+    sha256 = "5d968b0609f9bfbecc26753d21b5182e15183b672967f7dec6038404cd6a6d7f";
+  };
+  buildInputs = [ cmake fuse openssl zlib bzip2 libxml2 icu ];
+  cmakeConfigureFlagFlags = ["-DCMAKE_BUILD_TYPE=RELEASE"];
+  meta = {
+    homepage = http://www.darlinghq.org/;
+    description =
+    lib.strings.concatStringsSep " " ["Darling lets you open OS X dmgs on Linux"];
```
Comment (copumpkin) [inline pkgs/tools/filesystems/darling-dmg/default.nix]: Why the `concatStringsSep`?

File: pkgs/tools/filesystems/darling-dmg/default.nix [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    platforms = stdenv.lib.platforms.linux;
+    license = stdenv.lib.licenses.gpl3;
+  };
```
Comment (copumpkin) [inline pkgs/tools/filesystems/darling-dmg/default.nix]: Wanna add yourself as a maintainer? That'll let hydra notify you if it breaks and so on.

File: pkgs/tools/filesystems/darling-dmg/default.nix (hunk 4 above)
Comment (fxfactorial) [inline pkgs/tools/filesystems/darling-dmg/default.nix]: Ah, it was leftover junk, sorry.

File: pkgs/tools/filesystems/darling-dmg/default.nix (hunk 5 above)
Comment (fxfactorial) [inline pkgs/tools/filesystems/darling-dmg/default.nix]: Not really cause its not my package, I just needed it. but I will add myself to my ocaml packages in the future PRs.

File: pkgs/tools/filesystems/darling-dmg/default.nix [hunk 6]
```diff
@@ -0,0 +1,19 @@
+{ stdenv, fetchFromGitHub, lib, cmake, fuse, zlib, bzip2, openssl, libxml2, icu } :
+
+stdenv.mkDerivation rec {
+  name = "darling-dmg";
```
Comment (fpletz) [inline pkgs/tools/filesystems/darling-dmg/default.nix]: One more nitpick: This should have a version or a timestamp to make it updateable via nix-env.

Comment (fxfactorial) [inline pkgs/tools/filesystems/darling-dmg/default.nix]: Is there a check list for these kind of things or does everyone have to learn the nuances of these hassles?

**Comment by fxfactorial** [comment]
@fpletz I think this ought to be good now.

File: pkgs/tools/filesystems/darling-dmg/default.nix [hunk 7]
```diff
@@ -0,0 +1,20 @@
+{ stdenv, fetchFromGitHub, lib, cmake, fuse, zlib, bzip2, openssl, libxml2, icu } :
+
+stdenv.mkDerivation rec {
+  name = "darling-dmg-${version}";
+  version = "1.0";
+  src = fetchFromGitHub {
+    owner = "darlinghq";
+    repo = "darling-dmg";
+    rev = "9522e3907b82f6cde141b3e0e0063f09c8710cbf";
```
Comment (fpletz) [inline pkgs/tools/filesystems/darling-dmg/default.nix]: After looking at the upstream repo: There are tags available which should be used IMHO. This rev refers to version 1.0.3, not 1.0. In the meantime 1.0.4 was released btw. :)

**Comment by fxfactorial** [comment]
Never mind, this hassle is incredibly not worth it and the ":)" after I've spent so much time for minutiae is certainly not appreciated. I will recommend anyone who looks at nix to avoid it.

**Comment by copumpkin** [comment]
@fxfactorial sorry your experience has been bad with this, but we're all volunteers and I personally have had almost no time for anything Nix-related in the past couple of weeks. I'd urge you to not close the PR, as I don't think that @fpletz's point is worth blocking a merge over and I doubt he would block the merge either. Being one minor revision behind seems like a pretty minor issue in the grand scheme of things, but at the same time I think it's worth reminding people of good practices where possible.

**Comment by copumpkin** [comment]
I'm going to merge this for now, but also think we should address @fxfactorial's bigger points about our process to make this project more welcoming for newcomers. @fxfactorial if you're still willing to put a bit more effort against the project, I'd appreciate if you could write up a few points that caused you particular pain, but if not I'll also understand and I'll try to piece it together from your issue reports.

**Comment by copumpkin** [comment]
Dammit, I pressed the wrong button :smile:

**Comment by fxfactorial** [comment]
@copumpkin Okay, you're right. I spoke out of frustration sorry. I will give more energy to nix, I do want the project to succeed, I see it like a family member in the functional programming world.

**Comment by fpletz** [comment]
@fxfactorial I'm sorry if my comments disappointed and demotivated you. Believe me that I just wanted to help and point out some errors in good faith. I hope you keep learning & contributing to nix! I am certainly willing to continue helping you in any way possible! Thanks for your efforts!

**Comment by domenkozar** [comment]
My general rule is: when someone starts contributing, mention nitpicks and fix them myself. Once they submit a few PR, I expect those to be addressed.

**Comment by fpletz** [comment]
@domenkozar Thanks for the suggestion. That makes a lot of sense. I will do this from now on.

I've cleaned up the package.

**Pull request merged**